### DISCUSSION TITLE
Engine Preferences - enable engine toggle's visual representation is opposite actual state

### DISCUSSION THREAD

**Discussion by redxtech** [searxng/searxng#2591] [Q&A]
I'm configuring my enabled engines on my searxng instance, and when I have the toggle for certain engines set to `'on'`, it looks like the toggle is disabled. For whatever reason, the `'on'` state switch is greyed out and has an x on the toggle, and the `'off'` state shows a check mark, and the switch is coloured in. This seems highly counter intuitive, going against how toggle sliders should work.

When I save my settings, and look at both the cookie page to verify, and ones that appear enabled show up in the `disabled_engines` field. When I run a search, it shows that the results are from engines that visually appear to be disabled, with none from the engines that appear to be enabled.

I'm running `2023.8.3+203f1f092` (i recently upgraded from `2023.4`, but didn't get to check if this behavior was present before the upgrade), and I've checked on other instances of SearXNG and they appear to have the same issue, so it shouldn't be specific to mine.

It's my understanding that toggles such as these are supposed to represent their current state, rather than what will happen as a result of clicking them.

If this is intended behaviour, is there a reason why this is how it's displayed?

If not, how hard would a fix be to implement?

**Comment by redxtech** [comment]
I've actually done some more testing, and it looks like when the input is selected via inspect element, the `$0.value` is always `'on'`, regardless of the state of the slider.

This doesn't seem right either.

I've looked at a bunch of other settings, and it appears that engines are not the only settings this effects.

The enabled plugins, and I'm sure any other setting that used the "checkbox" slider would have the same issue.

**Comment by redxtech** [comment]
I have done some more testing with different browsers, and it appears that the issue was somehow on my end.

I had the same issue in both chrome and firefox, but when I opened up a new clean profile the toggles looked how I'd imagine they should.

I tried clearing cookies in my existing browser profiles, but that didn't do anything either.

What did end up fixing it was clearing all site data. I'm not sure why, or where this would have been stored, but it looks like after clearing all site data, it's working as it should.

If anyone knows what could have been causing this, that would be interesting to know.

**Comment by return42** [comment]
> enable engine toggle's visual representation is opposite actual state

AFAIK from user's side there is no issue (otherwise please open a bug report) / this issue is new to me, but I know the preferences implementation is a mess and needs to be re-implemented from ground / I started this task but haven't finished my work (its not such easy it looks at the first sight)

- https://github.com/searxng/searxng/issues/1440
